ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -626,10 +640,11 @@ private IndexValue markAsDeleted(StoreKey id, FileSpan fileSpan, MessageInfo inf
       newValue =
           new IndexValue(size, fileSpan.getStartOffset(), info.getExpirationTimeInMs(), info.getOperationTimeMs(),
               info.getAccountId(), info.getContainerId());
-      newValue.clearOriginalMessageOffset();
+      newValue.clearRelatedMessageOffset();
     } else {
-      newValue = new IndexValue(value.getSize(), value.getOffset(), value.getExpiresAtMs(), deletionTimeMs,
-          value.getAccountId(), value.getContainerId());
+      newValue =
+          new IndexValue(value.getSize(), value.getOffset(), value.getFlags(), value.getExpiresAtMs(), deletionTimeMs,
```

Comment:
you are making an explicit decision here to set `Ttl_Update_Index` flag in addition to `Delete_Index` flag for the delete entry if that happens after the TTL Update. Is this what we want? I understand that the Delete Entry keeps the expiration time, but just want to make sure you have thought this through.

Reply:
Yup - i want both flags set. There is no precedence on this since we ever had one flag but I think it is useful for the latest one to have _all_ the flags that have ever been set.